src/lune/builtins/luau.rs

```diff
@@ -0,0 +1,73 @@
+use mlua::prelude::*;
+use mlua::Compiler as LuaCompiler;
+
+use crate::lune::lua::table::TableBuilder;
+// use  as LuaNumber;
+
+pub fn create(lua: &'static Lua) -> LuaResult<LuaTable> {
+    TableBuilder::new(lua)?
+        .with_function("compile", compile_source)?
+        .with_function("load", load_source)?
+        .build_readonly()
+}
+
+fn compile_source<'a>(
+    lua: &'static Lua,
+    (source, options): (LuaString<'a>, Option<LuaTable<'a>>),
+) -> LuaResult<LuaString<'a>> {
+    let mut optimization_level = 1;
+    let mut coverage_level = 0;
+    let mut debug_level = 1;
+
+    if let Some(options) = options {
+        optimization_level = match options.raw_get("optimizationLevel")? {
+            LuaValue::Integer(val) => val as u8,
```

Comment:
We probably need to match on `LuaValue::Number(...)` too, integers don't actually exist in Luau and when `mlua` decides to use them or not is somewhat unclear...

Reply:
Addressed in; https://github.com/filiptibell/lune/pull/82/commits/34a44be3ff591625f171fc9c835a2b8bc1f5af63